orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -321,6 +321,14 @@ def _get_exported_model(self, input_schema: _ModelInputOutputSchemaType, *inputs
             # FlattenedModule needs _InputInfo to expand user input from *args to *args + **kwargs
             self._flattened_module._input_info = self._input_info
 
+            # Leverage cached model if available
+            cache_dir = self._runtime_options.ortmodule_cache_dir
+            if cache_dir and os.path.exists(cache_dir):
```

Comment:
it shtat possible we check whether model is changed or not by its hashing before we load from cache?

Reply:
The filename check will fail if the hash is different and save a new model in the same folder.

Reply:
Here we are only hashing the model architecture (and rightly so). We shouldn't check for model changes (like inputs or parameters), as that is a more involved process.

Reply:
model inputs is not a problem. Parameters might differ everytime it runs, if random seed is not fixed. Fintuning might be better as they just load weights from ckpts. 

Before we have a more comprehensive hash here, we have to be extremely careful the impact that this cache brings. 

BTW, shall we add a note somewhere like this https://github.com/microsoft/onnxruntime/blob/b7176f98266924053204fe25c3f2e701615c2585/orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py#L645, to inform cache is enabled or not.

Reply:
str(pt_model) returns the model architecture without the weights or inputs details.
So I think the hash of the model architecture might be a good candidate to save the file as.

Reply:
Adjusted filename sine python hashing is not guaranteed to be unique across sessions (ie. in distributed training scenario)